app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.kt

```diff
@@ -1,17 +1,105 @@
 package com.kickstarter.viewmodels
 
+import android.util.Pair
 import androidx.annotation.NonNull
 import com.kickstarter.libs.ActivityViewModel
+import com.kickstarter.libs.CurrentUserType
+import com.kickstarter.libs.Either
 import com.kickstarter.libs.Environment
+import com.kickstarter.libs.rx.transformers.Transformers
+import com.kickstarter.libs.utils.ProjectUtils
+import com.kickstarter.models.Project
+import com.kickstarter.models.Update
+import com.kickstarter.models.User
+import com.kickstarter.services.ApiClientType
+import com.kickstarter.ui.IntentKey
 import com.kickstarter.ui.activities.CommentsActivity
+import com.kickstarter.ui.data.ProjectData
+import rx.Observable
+import rx.android.schedulers.AndroidSchedulers
+import rx.subjects.BehaviorSubject
 
 interface CommentsViewModel {
 
     interface Inputs
-    interface Outputs
+    interface Outputs {
+        fun currentUserAvatar(): Observable<String?>
+        fun enableCommentComposer(): Observable<Boolean>
+        fun showCommentComposer(): Observable<Void>
+    }
 
     class ViewModel(@NonNull val environment: Environment) : ActivityViewModel<CommentsActivity>(environment), Inputs, Outputs {
+
+        val currentUser: CurrentUserType = environment.currentUser()
+        val client: ApiClientType = environment.apiClient()
         val inputs: Inputs = this
         val outputs: Outputs = this
+
+        private val currentUserAvatar = BehaviorSubject.create<String?>()
+        private val enableCommentComposer = BehaviorSubject.create<Boolean>()
+        private val showCommentComposer = BehaviorSubject.create<Void>()
+        init {
+
+            val loggedInUser = this.currentUser.loggedInUser()
+                    .filter { u -> u != null }
+                    .map { requireNotNull(it) }
+
+            loggedInUser
+                    .compose(bindToLifecycle())
+                    .subscribe {
+                        currentUserAvatar.onNext(it.avatar().small())
+            }
+
+            loggedInUser
+                    .compose(bindToLifecycle())
+                    .subscribe {
+                        showCommentComposer.onNext(null)
+                    }
+
+              intent()
+                    .map { it.getParcelableExtra(IntentKey.PROJECT_DATA) as ProjectData? }
+                    .ofType(ProjectData::class.java)
+                    .filter { it.project().isBacking || ProjectUtils.userIsCreator(it.project(), it.user()) }
```

Comment:
nit: `project().isBacking || ProjectUtils.userIsCreator(it.project(), it.user()` could be extracted to a function it's used in lunes 62, 66 and 96

Reply:
Done